_docs/technical/2017-01-15-cli-options.md

```diff
@@ -0,0 +1,727 @@
+Both directories must have one structure (e.g. they must contain the same files), otherwise `cardano-genupdate` will fail.
+
+Example of `cardano-genupdate` command:
+
+~~~
+stack exec -- cardano-genupdate /tmp/v000 /tmp/v001 /tmp/update.tar
+~~~
+
+**Important**: `cardano-genupdate` uses [`bsdiff`](http://www.daemonology.net/bsdiff/) program, so command `bsdiff` must be available in the `PATH`.
+
+## `cardano-wallet-hs2purs`
+
+An executable `cardano-wallet-hs2purs` is using for generating PureScript types based on Haskell types. We use it during buiding Daedalus wallet.
+
+`cardano-wallet-hs2purs` hasn't any options or parameters, it runs in Cardano SL root directory and produce `.purs` files in `daedalus/src/Generated` subdirectory.
+
+## `cardano-analyzer`
+
+An executable `cardano-analyzer` is...
+
+Available options:
+
+1.  `--version` and `--help`
+    <br/>Standard options.
+    <br/>&nbsp;
+2.  `--file FILEPATH`
+    <br/>Path to JSON file to analyze.
+    <br/>Specified path must exist.
+    <br/>**This option can be used more than once.**
+    <br/>Default value: none.
+    <br/>&nbsp;
+3.  `--tx-file FILEPATH`
+    <br/>Path to JSON file generated by `txSender`.
+    <br/>&nbsp;
+4.  `-k, --confirmation ARG`
+    <br/>How many blocks are needed for confirmation.
+    <br/>Default value: none.
+    <br/>&nbsp;
+
+## `cardano-avvmmigrate`
+
+An executable `cardano-avvmmigrate` is an AVVM converter. AVVM is a mechanism people buy ADA-certificates with. ADA-certificate is a special document with seed for generation of key pair. We're storing public key only and bind money to it. So `cardano-avvmmigrate` takes JSON file with "address + money" pairs and generates special binary file with that info. Based on this binary file the node knows genesis utxo.
+
+Part of input JSON file:
+
+~~~json
+{
+    "utxo": [{
+        "coin": {
+                "coinAmount": 413368,
+                "coinColor": {
+                        "getColor": 10
+                }
+        },
+        "address": "0TXMbbVfRFCfkWCklo9Qc3V71HlsTAQNQXDPryOE5kk="
+    }, {
+        "coin": {
+                "coinAmount": 421370,
+                "coinColor": {
+                        "getColor": 10
+                }
+        },
+        "address": "47gm-3YJrxBW4tROcNu5Mi6mfc4Fe8RxDgQjGaFIgEA="
+    ...
+}
+~~~
+
+Usage:
+
+~~~
+cardano-avvmmigrate PATH_TO_JSON BIN_OUTPUT_FILE CERT_INFO
+~~~
+
+where:
+
+1. `PATH_TO_JSON` - path to JSON file with AVVM data. Path must exist.
+2. `BIN_OUTPUT_FILE` - path to output binary file (recommend extension is `.bin`).
+3. `CERT_INFO` - certificates flag. Possible values - `nocerts` and `randcerts`.
+
+Example of `cardano-avvmmigrate` command:
+
+~~~
+stack exec -- cardano-avvmmigrate /tmp/utxo.json /tmp/avvm.bin nocerts
+~~~
+
+## `cardano-checks`
+
+An executable `cardano-checks` is an extractor of special check-comments. We need it to review all checks and verifications we have in all parts, such as receiving block/tx/whatever.
```

Comment:
"in all parts" of what? Could use some clarification. Also, "such as receiving block/tx/whatever". What does this refer to?